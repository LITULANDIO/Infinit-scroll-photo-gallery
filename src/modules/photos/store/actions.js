import axios from 'axios';
import photosApi from '@/api/api.js';

export const getPhotos = async () =>{
    let resp = [];
    try{
        const { data } = await photosApi.get();
        resp = data;
    }catch(error){
        console.error(error)
    }
    return resp
}