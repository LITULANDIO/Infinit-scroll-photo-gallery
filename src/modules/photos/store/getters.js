export const getPhotos = (state) =>{
    return state.photos;
}