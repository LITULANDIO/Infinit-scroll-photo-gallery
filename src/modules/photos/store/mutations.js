export const SET_PHOTOS = (state, payload) =>{
    state.photos = payload
}