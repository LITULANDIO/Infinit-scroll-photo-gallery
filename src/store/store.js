import { createStore } from "vuex";
import createPersistedState from "vuex-persistedstate";

const store = createStore({
   state:{
   },
   getters:{

   },
   mutations:{

   },
   actions:{

   },
  plugins: [createPersistedState()],

})

export default store