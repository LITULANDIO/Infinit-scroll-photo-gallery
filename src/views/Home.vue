<template>
<!-- <transition name="fade">
    <div v-if="show" class="w-full h-screen | mt-20 | fixed | bg-gray-600 opacity-100"></div>
</transition> -->
<div class="pt-16">
  <div class="flex justify-center items-center flex-wrap">
      <Photo :photo="photo" v-for="photo in photos" :key="photo.id" @onClicked="onDelete"/>
  </div>
</div>
      <Spinner :show="show" :load="countPhotos"/>
</template>

<script>
import usePhotos from '@/modules/photos/composable/usePhotos.js';
import Photo from '@/components/photo/Photo.vue';
import Spinner from '@/components/spinner/Spinner.vue';
import { onMounted, onUnmounted, onBeforeUpdate, ref, watchEffect } from 'vue';

export default {
    name: 'Home',
    components:{
        Photo,
        Spinner
    },
    setup(){
    const { limitScreenPhotos, deletePhoto, photos } = usePhotos();
    let num = 66;
    const show = ref(false);
    const countPhotos = ref(0)
       
    const loadPhotos = async() =>{
        photos.value = await limitScreenPhotos(num);
    }      
    const moreLoadPhotos = async() =>{
         photos.value = await limitScreenPhotos(num+= 66);
    }
  
    onMounted(() =>{
        loadPhotos();
        window.addEventListener('scroll', handleScroll);
        window.addEventListener('touchend', handleScroll);

    });
    onUnmounted(() =>{
        window.removeEventListener('scorll', handleScroll);
        window.addEventListener('touchend', handleScroll);

    })
    onBeforeUpdate(() =>{
        if(show.value){
        setTimeout(() => show.value = false, 500)
        }
        if(photos.value){
            countPhotos.value = photos.value.length
        }
    })

  
    const handleScroll = (event) =>{
        if(document.body.scrollHeight - window.innerHeight === window.scrollY){
            show.value = true;
            moreLoadPhotos()
        }
    }
    const onDelete = (photo) =>{
        deletePhoto(photo);
    }


        return{
            photos,
            moreLoadPhotos,
            show,
            onDelete,
            countPhotos
        }
    }

}
</script>

<style lang="scss" scoped>
.fade-enter-from{
    opacity: 0;
}
.fade-enter-to{
    opacity: 1;
}
.fade-enter-active{
    transition: all 1s ease;
}
.fade-leave-from{
    opacity: 1;
}
.fade-leave-to{
    opacity: 0;
}
.fade-leave-active{
    transition: all 1s ease;
}

</style>