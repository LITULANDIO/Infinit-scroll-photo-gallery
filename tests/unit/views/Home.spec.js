import { shallowMount } from '@vue/test-utils';
import { createStore } from "vuex"
import photos from '@/modules/photos/store';
import { photosState } from '../mocks/store.mock';
import Home from '@/views/Home.vue';

const createVuexStore = (initialState) => 
    createStore({
        modules:{
            photos:{
                ...photos,
                state: { ...initialState }
            }
        }
    })


describe('Home component', () =>{

    const store = createVuexStore(photosState);
    let wrapper;

    // beforeEach(() =>{
    //     jest.clearAllMocks()
    //     wrapper = shallowMount(Home,{
    //         plugins: [store]
    //     })
    // })

    test('must match snapshot', async () =>{
        const wrapper = shallowMount(Home)
        expect(wrapper.html()).toMatchSnapshot()

        // check loading photos
        expect(await wrapper.vm.moreLoadPhotos(66));

        //check components home
        expect(wrapper.find('spinner-stub').exists()).toBeTruthy();
        expect(wrapper.find('photo-stub').exists()).toBeTruthy();

    })

    
    
})
