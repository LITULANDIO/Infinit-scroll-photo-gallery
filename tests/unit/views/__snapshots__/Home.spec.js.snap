// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component must match snapshot 1`] = `
<transition-stub>
  <!--v-if-->
</transition-stub>
<div class="pt-16">
  <div class="flex justify-center items-center flex-wrap"></div>
</div>
<spinner-stub show="false"></spinner-stub>
`;
